### PR TITLE
Add OpenCorporates.php to RecordCollectors.

### PR DESCRIPTION
**Pull request by rembiesa**
image

### CODE REVIEW

File: app/RecordCollectors/OpenCorporates.php [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	public $label = 'LBL_OC';
+
+	/** {@inheritdoc} */
+	public $displayType = 'FillFields';
+
+	/** {@inheritdoc} */
+	public $description = 'LBL_OC_DESC';
+
+	/** {@inheritdoc} */
+	public $docUrl = 'https://api.opencorporates.com/documentation/API-Reference';
```
Comment (mariuszkrzaczkowski) [inline app/RecordCollectors/OpenCorporates.php]:
```suggestion
	public $docUrl = 'https://api.opencorporates.com';
```

File: app/RecordCollectors/OpenCorporates.php [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+					$response['errorCode'] = $e->getCode();
+					$response['error'] = \App\Language::translate('LBL_OC_403', 'Other.RecordCollector');
+					break;
+				case 404:
+					$response['errorCode'] = $e->getCode();
+					$response['error'] = \App\Language::translate('LBL_OC_404', 'Other.RecordCollector');
+					break;
+				default:
+					$response['errorCode'] = $e->getCode();
+					$response['error'] = $e->getMessage();
```
Comment (mariuszkrzaczkowski) [inline app/RecordCollectors/OpenCorporates.php]:
```suggestion
					$response['error'] = $e->getResponse()->getReasonPhrase();
```

**Review by mariuszkrzaczkowski — CHANGES_REQUESTED** [review]

File: app/RecordCollectors/OpenCorporates.php [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+	 *
+	 * @return void
+	 */
+	public function getCodesFromApi(array $data)
+	{
+		try {
+			$response = \App\Json::decode(\App\RequestHttp::getClient()->get($this->url . self::API_VERSION . '/jurisdictions')->getBody()->getContents());
+		} catch (\GuzzleHttp\Exception\ClientException $e) {
+			\App\Log::warning($e->getMessage(), 'RecordCollectors');
+			$this->response['error'] = $e->getMessage();
```
Comment (mariuszkrzaczkowski) [inline app/RecordCollectors/OpenCorporates.php]: if ($e->getCode() > 400) {
				$this->response['error'] = $e->getResponse()->getReasonPhrase();
			} else {
				$this->response['error'] = $e->getMessage();
			}

File: app/RecordCollectors/OpenCorporates.php [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+
+		$this->data = $response;
+	}
+
+	/**
+	 * Function fetching and setting codes (countries, districts) for&from OpenCorporates.
+	 *
+	 * @param array $data
+	 *
+	 * @return void
```
Comment (mariuszkrzaczkowski) [inline app/RecordCollectors/OpenCorporates.php:176]:
```suggestion
	 * @return array
```

**Comment by rembiesa** [comment]
too small daily query limit, we are abandoning the implementation.

**Pull request closed without merging**